README.md

```diff
@@ -478,3 +478,11 @@ At the bottom of the workflow summary page, there is a dedicated section for art
 There is a trashcan icon that can be used to delete the artifact. This icon will only appear for users who have write permissions to the repository.
 
 The size of the artifact is denoted in bytes. The displayed artifact size denotes the size of the zip that `upload-artifact` creates during upload.
+
+# Recommended Permissions
+
+The `actions/upload-artifact` workflow relies on an internal authentication pattern and does not use the GITHUB_TOKEN, to reduce risk of over-privileged token, jobs that use `actions/upload-artifact` should set permissions to none:
```

Comment:
Same feedback as cache and download-artifact - https://github.com/actions/download-artifact/pull/373#discussion_r1953137926

Reply:
I think best practice is to set permissions at the top to minimal (none or contents: read for example), and then add permissions needed in each job.

See the wording in "remediation steps" for example on the ossf scorecard: https://github.com/ossf/scorecard/blob/7ce8609469289d5f3b1bf5ee3122f42b4e3054fb/docs/checks.md#token-permissions

Agreed that the proposed wording without links to more detailed documentation might break things for people. 

Some docs explaining the permissions required would be good, and pointers to more detailed documentation.